src/algorithms/calorimetry/CalorimeterHitReco.cc

```diff
@@ -208,21 +208,31 @@ void CalorimeterHitReco::AlgorithmProcess() {
 //                dd4hep::Position(gpos.x(), gpos.y(), gpos.z()));//dd4hep::Position(gpos.x, gpos.y, gpos.z)
         std::vector<double> cdim;
         // get segmentation dimensions
-        if (converter->findReadout(local).segmentation().type() != "NoSegmentation") {
-            cdim = converter->cellDimensions(cellID);
+        auto segmentation_type = converter->findReadout(local).segmentation().type();
+        auto cell_dim = converter->cellDimensions(cellID);
+        if (segmentation_type == "CartesianGridXY") {
+            cdim.resize(3);
+            cdim[0] = cell_dim[0];
+            cdim[1] = cell_dim[1];
+            m_log->error("Using segmentation for cell dimensions: {}", fmt::join(cdim, ", "));
         } else {
+            if (segmentation_type != "NoSegmentation") {
+                m_log->warn("Usupported segmentation type \"{}\"", segmentation_type);
+            }
+
             // Using bounding box instead of actual solid so the dimensions are always in dim_x, dim_y, dim_z
             cdim = converter->findContext(cellID)->volumePlacement().volume().boundingBox().dimensions();
             std::transform(cdim.begin(), cdim.end(), cdim.begin(),
                            std::bind(std::multiplies<double>(), std::placeholders::_1, 2));
+            m_log->error("Using bounding box for cell dimensions: {}", fmt::join(cdim, ", "));
```

Comment:
Were these `error`s intended to be `debug`s? Physics benchmark now looks like https://eicweb.phy.anl.gov/EIC/benchmarks/physics_benchmarks/-/jobs/1659794 ...